[Node.js] Style guide fixes for TDBX vale rules (#842)
mongodb/docs-node: source/fundamentals/authentication/mechanisms.txt

   * - ``tlsAllowInvalidHostnames``
     - boolean
     - ``false``

>      - Specifies whether the driver will permit a mismatch between the
>        server hostname and TLS certificate hostname.

Author: I went with a slightly different change. Using "whether" keeps it parallel with the other entries on the page. Good call out on the error being raised.

```suggestion
     - Specifies whether the driver raises an error when there is a mismatch between the
       server hostname and TLS certificate hostname.
```
